### PR TITLE
ref/consolidate_DemoMenuItem_with_AdType

### PR DESCRIPTION
**Pull request by angelwu0409**
Have just changed the Java project, not sure if this is the best practice to consolidate these 2 menus items. 
Will change the Kotlin part of the project after :)

### CODE REVIEW

File: AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/data/main/DemoMenuItem.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    {
+        return runnable;
+    }
+
+    @Override
+    public int getType()
+    {
+        return TYPE_AD_ITEM;
+    }
+}
```
Comment (thomasmso) [inline AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/data/main/DemoMenuItem.java:46]: nl at end of file is convention

File: AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/data/main/DemoMenuItem.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        runnable = null;
+
+    }
+
+    public DemoMenuItem(final String title, final Runnable runnable)
+    {
+        this.title = title;
+        this.intent = null;
+        this.runnable = runnable;
+
```
Comment (thomasmso) [inline AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/data/main/DemoMenuItem.java:25]:
```suggestion
```

File: AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/data/main/DemoMenuItem.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    private final String title;
+    private final Intent intent;
+    private final Runnable runnable;
+
+    public DemoMenuItem(final String title, final Intent intent)
+    {
+        this.title = title;
+        this.intent = intent;
+        runnable = null;
+
```
Comment (thomasmso) [inline AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/data/main/DemoMenuItem.java:17]:
```suggestion
        this.runnable = null;
```

File: AppLovin MAX Demo App - Kotlin/app/src/main/java/com/applovin/enterprise/apps/demoapp/ads/max/mrecs/MrecAdActivity.kt [hunk 4]
```diff
@@ -2,7 +2,7 @@ package com.applovin.enterprise.apps.demoapp.ads.max.mrecs
 
 import android.content.Intent
 import com.applovin.enterprise.apps.demoapp.ads.DemoMenuActivity
-import com.applovin.enterprise.apps.demoapp.ads.DemoMenuItem
+import com.applovin.enterprise.apps.demoapp.data.main.DemoMenuItem
```
Comment (angelwu0409) [inline AppLovin MAX Demo App - Kotlin/app/src/main/java/com/applovin/enterprise/apps/demoapp/ads/max/mrecs/MrecAdActivity.kt:5]: Seems like the Mrec activity actually is not working as expected, not sure what exactly is different for MERC and other activities, might look into it after:
    java.lang.RuntimeException: Unable to start activity ComponentInfo{com.applovin.enterprise.apps.demoapp/com.applovin.enterprise.apps.demoapp.ads.max.mrecs.ProgrammaticMrecAdActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'void androidx.recyclerview.widget.RecyclerView.setHasFixedSize(boolean)' on a null object reference

Comment (thomasmso) [reply AppLovin MAX Demo App - Kotlin/app/src/main/java/com/applovin/enterprise/apps/demoapp/ads/max/mrecs/MrecAdActivity.kt:5]: Feel free to fix :)

**Pull request merged**